python/sglang/srt/layers/moe/fused_moe_triton/layer.py

```diff
@@ -97,22 +110,23 @@ def create_weights(
         **extra_weight_attrs,
     ):
         # Fused gate_up_proj (column parallel)
-        w13_weight = torch.nn.Parameter(
-            torch.empty(
-                num_experts, 2 * intermediate_size, hidden_size, dtype=params_dtype
-            ),
-            requires_grad=False,
+        w13_weight_raw = torch.empty(
+            num_experts, 2 * intermediate_size, hidden_size, dtype=params_dtype
         )
+        if self.use_triton_kernels:
+            w13_weight_raw = w13_weight_raw.transpose(-2, -1).contiguous()
+        w13_weight = torch.nn.Parameter(w13_weight_raw, requires_grad=False)
+
```

Comment:
```suggestion
        w13_weight_n, w13_weight_k = 2 * intermediate_size, hidden_size
        if self.use_triton_kernels:
            w13_weight_n, w13_weight_k = w13_weight_k, w13_weight_n
        w13_weight = torch.nn.Parameter(
            torch.empty(
                num_experts, w13_weight_n, w13_weight_k, dtype=params_dtype
            ),
            requires_grad=False,
        )
```

Reply:
Using contiguous will trigger tensor allocation, better to avoid this.

Reply:
Make sense. Fixed.